### ISSUE TITLE
Is this a bug ? Can't set value all the time

### ISSUE THREAD

**Issue by Nightshades1** [frida/frida#292]
Hello i'm hacking on a game and i'm encountering an issue with frida (REPEL).

This is looking like that

Interceptor.attach(Module.findExportByName("Library.so", "TheFunction"), {
onEnter: function (args) {
args[1] = ptr(1);
console.log("Function entered ! " + " args 1 " + args[1].toInt32());
},
onLeave: function (retval) {
console.log('function leave : ' + retval.toInt32());
}
});

So i got this from the REPEL Function entered !  args 1 1 which is fine
but then come the next value (read from the player) and it is Function entered !  args 1 13602 (13602 Health).

I did even test with conditional statement like if(args[1] == 1200)
{
args[1] = ptr(1);
}

But it is only executed one time then when there is the next value it doesn't work anymore.
It's like Frida simply do it once and end of story, did i do something wrong ?

Anyway i used frida for years it's the best mobile hacking framework 🗡 .
Android 4.4.4 armv7

TL;DR
When a function is executed ONE TIME it will change the value one time then the next value being readt won't be.

**Comment by oleavr** [comment]
Hi,

Which version of Frida is this on? There used to be a bug where only the first assignment would happen, but that was fixed a long time ago. By the way, instead of:
`if (args[1] == 1200)`
do:
`if (args[1].equals(1200))`
(And when you want to compare directly, always use `===` and `!==`.)

Cheers!
